src/shared/bridge.js

```diff
@@ -78,46 +76,54 @@ export default class Bridge {
    * callback when all results are collected.
    *
    * @param {string} method - Name of remote method to call.
-   * @param {any[]} args - Arguments to method.
+   * @param {any[]} args - Arguments to method. Final argument is an optional
+   *   callback with this type: `(error: string|Error, ...result: any[]) => void`
    * @return {Promise<any[]>} - Array of results, one per connected frame
    */
   call(method, ...args) {
     let cb;
-    if (typeof args[args.length - 1] === 'function') {
-      cb = args[args.length - 1];
+    const finalArg = args[args.length - 1];
+    if (typeof finalArg === 'function') {
+      cb = finalArg;
       args = args.slice(0, -1);
     }
 
-    const _makeDestroyFn = c => {
+    /** @param {RPC} channel */
+    const _makeDestroyFn = channel => {
       return error => {
-        c.destroy();
-        this.links = Array.from(this.links)
-          .filter(l => l.channel !== c)
-          .map(l => l);
+        channel.destroy();
+        this.links = this.links.filter(
+          registeredChannel => registeredChannel !== channel
+        );
         throw error;
       };
     };
 
-    const promises = this.links.map(l => {
-      const p = new Promise((resolve, reject) => {
+    const promises = this.links.map(channel => {
+      const promise = new Promise((resolve, reject) => {
         const timeout = setTimeout(() => resolve(null), 1000);
+        // This try/catch is only used on tests, to exercise the `_makeDestroyFn`
+        // function. In real life, the execution of the channel.call's callback will
+        // never throws an exception. The try/catch can be removed when we support
```

Comment:
This assumes that the `window.postMessage` or `messagePort.postMessage` call never throws. Are you sure  that's true?

Reply:
It is possible to throws but this try/catch will never catch it. The reason is that the callback is executive somewhere else in `RPC`.

Reply:
> I only found one way to make the postMessage to fail:

From looking at [the `postMessage` specification](https://html.spec.whatwg.org/multipage/web-messaging.html#message-port-post-message-steps) it looks to me like one thing that will cause an immediate `postMessage` failure is if the arguments to `postMessage` cannot be serialized. This would most likely indicate a bug in the calling code (eg. trying to invoke `call` with a non-serializable type of data.). I wonder if circular references can also cause a problem here?

Reply:
I found a bunch at least a couple more scenarios where `postMessage` could raise exception. For example:

```
window.postMessage([],'e.com', [])
Uncaught DOMException: Failed to execute 'postMessage' on 'Window': Invalid target origin 'e.com' in a call to 'postMessage'.

window.postMessage([],'http://e.com', [])
Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('http://e.com') does not match the recipient window's origin ('https://html.spec.whatwg.org').
```

I know understand that this is more common/plausible that I thought, so I will remove the comment.

Reply:
@esanzgar - Do these exceptions occur synchronously? ie. If you surround the `postMessage` with a try..catch does it get invoked. Sometimes in the browser console you can trigger an error which appears to have been reported synchronously but has in fact been reported via some async mechanism (eg. an 'error' event).

Reply:
The exception happens synchronously and it is intercepted by a surrounded `try/catch`.